compiler/jsgen.nim

```diff
@@ -2598,6 +2598,13 @@ proc genHeader(): Rope =
     var framePtr = null;
     var excHandler = 0;
     var lastJSError = null;
+    var globalThis;
+    if(typeof globalThis == 'undefined') globalThis = (function () { 
```

Comment:
When would this not be true? You start with ``var globalThis;``

Reply:
var a; typeof a  == "undefined";

Reply:
Exactly, you just wrote ``if (true)``. I think in order to emulate ``globalThis`` you must not predeclare it at all.

Reply:
a= 1; var a; typeof a == "undefined";  it's not like you think.

Reply:
declare it here just note it as global variable, it can be safely delete.